### PR TITLE
Add some options + compile fix.

### PR DESCRIPTION
**Pull request by hoskeri**
- add output file option.
- move reduce phase earlier - scala compiler
  wanted to add it to after the terminal phase.

### CODE REVIEW

**Comment by ngocdaothanh** [comment]
One problem is `emptyOutputFileExists` is removed in this pull request.

Currently, it acts as a signal to skip extracting i18n messages during compilation. If we remove it, Scala compilation will be slowed down (one big reason people hate Scala is because of its slow compilation).

So I think you should modify your pull request to keep `emptyOutputFileExists`.

File: src/main/scala/scala/Xgettext.scala [hunk 1]
```diff
@@ -30,7 +30,6 @@ class Xgettext(val global: Global) extends Plugin with ScalaVersionAdapter {
   val description = "This Scala compiler plugin extracts and creates gettext.pot file"
   val components  = List[PluginComponent](MapComponent, ReduceComponent)
 
-  val OUTPUT_FILE     = "i18n.pot"
   val HEADER          = """msgid ""
```
Comment (ngocdaothanh) [inline src/main/scala/scala/Xgettext.scala]: Also remove spaces between `HEADER` and `=`.

Comment (hoskeri) [inline src/main/scala/scala/Xgettext.scala]: Thank you for the review!

Ok - Does it really make the compiler much slower - any build system would try to compile only if the source changed, right? However, I could add a force option, too - would that work?

-Abhijit

**Comment by ngocdaothanh** [comment]
Can you fix the conflict with latest source code in master branch? And can you also improve your source code to address my previous comments?

**Comment by ngocdaothanh** [comment]
Declined this PR due to inactivity for a long time.

**Pull request closed without merging**